src/main/java/frc/robot/commands/CmdDriveToPosition.java

```diff
@@ -0,0 +1,76 @@
+package frc.robot.commands;
+
+import PIDControl.PIDControl;
+import PIDControl.PIDControl.Coefficient;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.drive.Drive;
+import frc.robot.subsystems.drive.Vector;
+
+public class CmdDriveToPosition extends CommandBase 
+{
+    private PIDControl _xPID;
+    private PIDControl _yPID;
+    private PIDControl _rotatePID;
+
+    private Vector _targetPosition;
+    private double _targetHeading;
+
+    public CmdDriveToPosition(Vector targetPosition, double targetHeading) 
+    {
+        _xPID.setCoefficient(Coefficient.P, 0.0, 0.015, 0.0);// based on previous code
+        _xPID.setCoefficient(Coefficient.I, 5.0, 0.0, 0.001);
+        _xPID.setCoefficient(Coefficient.D, 0.0, 0, 0.0);
+        _xPID.setInputRange(-720.0, 720.0);// assumed unit of inches
+        _xPID.setOutputRange(-1.0, 1.0);
+        _xPID.setOutputRamp(0.1, 0.05);
+        _xPID.setSetpointDeadband(2.0);
+
+        _yPID.setCoefficient(Coefficient.P, 0.0, 0.015, 0.0);// these should all be the same as xPID
+        _yPID.setCoefficient(Coefficient.I, 5.0, 0.0, 0.001);
+        _yPID.setCoefficient(Coefficient.D, 0.0, 0, 0.0);
+        _yPID.setInputRange(-720.0, 720.0);// assumed unit of inches
+        _yPID.setOutputRange(-1.0, 1.0);
+        _yPID.setOutputRamp(0.1, 0.05);
+        _yPID.setSetpointDeadband(2.0);
+
+        _rotatePID.setCoefficient(Coefficient.P, 0.005, 0.0, 0.0);
+        _rotatePID.setCoefficient(Coefficient.I, 20.0, 0.0, 0.00027);
+        _rotatePID.setCoefficient(Coefficient.D, 0.0, 0.0, 0.0);
+        _rotatePID.setInputRange(-180.0, 180.0);
+        _rotatePID.setContinuous(true);
+        _rotatePID.setOutputRamp(0.1, 0.05);
+        _rotatePID.setSetpointDeadband(2.0);
+    }
+
+    @Override
+    public void initialize() 
+    {        
+        _xPID.setSetpoint(_targetPosition.getX(), Drive.getInstance().getFieldPosition().getX(), true);
+        _yPID.setSetpoint(_targetPosition.getY(), Drive.getInstance().getFieldPosition().getY(), true);
```

Comment:
There's a ton of places we use the Drive subsystem instance in this class. We can shorten up some code if we create a member variable for the instance and get the instance once. Something like this:

```java
public class CmdDriveToPosition extends CommandBase
{
    private final Drive _drive = Drive.getInstance();

    // other methods
}
```

Then, these lines (and anywhere else we're using the subsystem) can shorten up a bit:
```suggestion
        _xPID.setSetpoint(_targetPosition.getX(), _drive.getFieldPosition().getX(), true);
        _yPID.setSetpoint(_targetPosition.getY(), _drive.getFieldPosition().getY(), true);
```